### PR TITLE
Deser header

### PR DESCRIPTION
**Pull request by RCasatta**
~~include #7~~

### CODE REVIEW

File: src/types.rs [hunk 1]
```diff
@@ -189,8 +200,8 @@ pub struct GetMerkleRes {
 pub struct HeaderNotification {
     /// New block height.
     pub height: usize,
-    #[serde(rename(serialize = "hex"))]
     /// Newly added header.
+    #[serde(rename(serialize = "hex", deserialize = "hex"), deserialize_with = "from_hex_header")]
```
Comment (afilini) [inline src/types.rs]: if both `serialize` and `deserialize` are the same I guess you could use [`rename = "hex"`](https://serde.rs/field-attrs.html#rename)

File: src/types.rs [hunk 2]
```diff
@@ -107,6 +107,17 @@ where
     Ok(answer)
 }
 
+fn from_hex_header<'de, D>(deserializer: D) -> Result<block::BlockHeader, D::Error>
```
Comment (afilini) [inline src/types.rs]: i might be missing something here, but aren't those types like `BlockHeader` already de-serializable? why did you have to re-implement the serializer manually?

Comment (RCasatta) [reply src/types.rs]: It just didn't work out of the box

File: src/types.rs (hunk 1 above)
Comment (RCasatta) [reply src/types.rs]: fixed

File: src/types.rs (hunk 2 above)
Comment (RCasatta) [reply src/types.rs]: it just doesn't work out of the box for me

Comment (RCasatta) [reply src/types.rs]: maybe because BlockHeader is missing FromStr

**Pull request merged**